src/algebra/homology/image_to_kernel.lean

```diff
@@ -213,21 +221,43 @@ def homology.map (p : α.right = β.left) :
 cokernel.desc _ (kernel_subobject_map β ≫ cokernel.π _)
   begin
     rw [image_subobject_map_comp_image_to_kernel_assoc w w' α β p],
-    simp,
+    simp only [cokernel.condition, comp_zero],
   end
 
 @[simp, reassoc]
 lemma homology.π_map (p : α.right = β.left) :
   homology.π f g w ≫ homology.map w w' α β p = kernel_subobject_map β ≫ homology.π f' g' w' :=
-by { simp [homology.π, homology.map], }
+by simp only [homology.π, homology.map, cokernel.π_desc]
 
 @[simp, reassoc]
 lemma homology.map_desc (p : α.right = β.left)
   {D : V} (k : (kernel_subobject g' : V) ⟶ D) (z : image_to_kernel f' g' w' ≫ k = 0) :
   homology.map w w' α β p ≫ homology.desc f' g' w' k z =
     homology.desc f g w (kernel_subobject_map β ≫ k)
-      (by simp [image_subobject_map_comp_image_to_kernel_assoc w w' α β p, z]) :=
-by { ext, simp, }
+      (by simp only [image_subobject_map_comp_image_to_kernel_assoc w w' α β p, z, comp_zero]) :=
+by ext; simp only [homology.π_desc, homology.π_map_assoc]
+
+@[simp]
+lemma homology.map_id : homology.map w w (𝟙 _) (𝟙 _) rfl = 𝟙 _ :=
+by ext; simp only [homology.π_map, kernel_subobject_map_id, category.id_comp, category.comp_id]
+
+@[reassoc]
+lemma homology.map_comp (p₁ : α₁.right = β₁.left) (p₂ : α₂.right = β₂.left) :
+  homology.map w₁ w₂ α₁ β₁ p₁ ≫ homology.map w₂ w₃ α₂ β₂ p₂ =
+    homology.map w₁ w₃ (α₁ ≫ α₂) (β₁ ≫ β₂)
+      (by simp only [comma.comp_left, comma.comp_right, p₁, p₂]) :=
```

Comment:
Can we introduce a separate lemma for this `by simp only ...` parameter?

Reply:
Done